toolkit/tools/internal/pkggraph/pkggraph_test.go

```diff
@@ -245,14 +265,7 @@ func checkTestGraph(t *testing.T, g *PkgGraph) {
 		pkgD2Unresolved,
 		pkgD3Unresolved,
 	}
-	for _, mustHave := range component1 {
-		found := false
-		for _, n := range g.AllNodesFrom(a.RunNode) {
-			found = found || mustHave.Equal(n)
-		}
-		assert.True(t, found)
-	}
-	assert.Equal(t, len(component1), len(g.AllNodesFrom(a.RunNode)))
+	checkEqualComponents(t, component1, g.AllNodesFrom(a.RunNode))
```

Comment:
I think we're not creating clones of the test nodes anyway, so working on pointers should be fine. If that's the case, won't this work instead of `checkEqualComponents`?
```suggestion
	assert.ElementsMatch(t, component1, g.AllNodesFrom(a.RunNode))
```

Reply:
Unfortunately not, they underlying objects are completely separate graph objecst so a custom equality check is needed (we use `PkgNode.Equal()`)